### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by carlos-ssh**
In this milestone we accomplished the following:

-Unit tests with Jest for the Operation.js and Calculate.js files for the Calculator.
-Unit test for each component pages to check the correct display message

Installation:

-npm install
-npm run tests

### CODE REVIEW

File: src/tests/operate.test.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+it('Test for the division operation (50%100=50)', () => {
+  const mockState = {
+    total: '50',
+    next: '100',
+    operation: '%',
+  };
+
+  expect(operate(mockState.total, mockState.next, mockState.operation)).toBe('50');
+});
```
Comment (Emmyn5600) [inline src/tests/operate.test.js]: - [x] Please start with `describe` to use describe allows you to gather your tests into separate groupings within the same file, even multiple nested levels

- [x] Good job here, however, let us add more tests to move forward.

- Should throw an error if given a wrong number
- Should give a 0 error if given a wrong operator

File: src/tests/calculate.test.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  };
+
+  const newValue = '.';
+
+  expect(calculate(mockState, newValue)).toStrictEqual({
+    total: '1',
+    next: '5.',
+    operation: null,
+  });
+});
```
Comment (Emmyn5600) [inline src/tests/calculate.test.js]: - [x] Please start with describing, describe allows you to gather your tests into separate groupings within the same file, even multiple nested levels

- [ ] Implement the following tests.

- should throw 0 error if trying to make a zero division
- should throw Zero division error

File: src/tests/pages/pages.test.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  it('Should render CalculatorPage', async () => {
+    render(<CalculatorPage />);
+    expect(screen.getByText('Lets do some math!')).toBeInTheDocument();
+  });
+
+  it('Should render QuotePage', async () => {
+    render(<QuotePage />);
+    expect(screen.getByText('Math Quote..')).toBeInTheDocument();
+  });
+});
```
Comment (Emmyn5600) [inline src/tests/pages/pages.test.js]: - [x] There should be the usage of Jest snapshots for testing all the components.

After usage of Jest snapshots, the snapshot folder will be created with the components that have been tested

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @RafaelEchart  and @carlos-ssh 

Great work on making the project requirements 👏 
You've done well implementing some of the project requirements, but there are still some which aren't addressed yet, please work on it and ask for another review.


 ## Required Changes ♻️
 
_Check the comments under the review._

## Optional suggestions

- N/A 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: src/tests/operate.test.js (hunk 1 above)
Comment (RafaelEchart) [reply src/tests/operate.test.js]: Thanks for your comment!
Please some of your suggestions don't have any sense, please try to check before submitting 
Example:
'what is happening at next'
'let us add more tests'

File: src/tests/calculate.test.js (hunk 2 above)
Comment (RafaelEchart) [reply src/tests/calculate.test.js]: Thanks for the comment!

I hope I did not get this wrong but the two bullet points are about the same issue. Please be more clear.

File: src/tests/pages/pages.test.js (hunk 3 above)
Comment (RafaelEchart) [reply src/tests/pages/pages.test.js]: I like this one! Thanks for the suggestions I had a very hard time trying to figure it out what was this snapshot.

I used **'react-test-renderer'**

Thanks!

**Review by ArthurGC — APPROVED** [review]
# Approved project!  💯 ✅ 🥳 🎈 🎉
Hi  @carlos-ssh and @RafaelEchart 

Amazing job implementing suggestions from the previous reviewer! 💯 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 




Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/tests/operate.test.js (hunk 1 above)
Comment (Emmyn5600) [reply src/tests/operate.test.js]: Not sure what you mean by it doesn’t make sense, kindly take a look in this [resource](https://www.xoriant.com/blog/product-engineering/jest-fast-furious-unit-testing-framework.html), you will understand what I meant

**Pull request merged**